x-pack/platform/packages/shared/onechat/onechat-common/chat/conversation.ts

```diff
@@ -154,13 +160,28 @@ export const isReasoningStep = (step: ConversationRoundStep): step is ReasoningS
  */
 export type ConversationRoundStep = ToolCallStep | ReasoningStep;
 
+export enum ConversationRoundStatus {
+  /** round is currently being processed */
+  inProgress = 'in_progress',
+  /** the round is completed */
+  completed = 'completed',
+  /** round has been interrupted and is awaiting user input */
+  awaitingPrompt = 'awaiting_prompt',
+}
+
 /**
  * Represents a round in a conversation, containing all the information
  * related to this particular round.
  */
 export interface ConversationRound {
   /** unique id for this round */
   id: string;
+  /** current status of the round */
+  status: ConversationRoundStatus;
+  /** persisted state to resume interrupted states */
+  state?: RoundState;
+  /** if status is awaiting_prompt, contains the current prompt request*/
+  pending_prompt?: PromptRequest;
```

Comment:
I thought about having a `prompt_request` round step instead, but in practice prompt requests happens from within tool steps, so I ended up storing it at the top level of the round right now.

It may not be sufficient (if we want to show those prompts in the history for example), but I want to take some additional time to think about it, and I think it's fine adding it / revisiting as a follow-up, as it doesn't change much the execution flow (which was the most tricky part of the PR)